### DISCUSSION TITLE
Node Public Interfaces Team Weekly Update - 2021-07-16

### DISCUSSION THREAD

**Discussion by frol** [near/nearcore#4537] [Node Public Interfaces]
We have not had an update last week, so the previous update is from two weeks ago #4454, and this update includes all the notes from the last two weeks.

# Brief Summary

Since the end of June we defined the team OKRs for 2021-Q3, and work through them. These last week we were mostly focused on small improvements to the stability of the network, reliability of implementation, and improvements to documentation.

* **Heads up: 1.20.0 release on mainnet was re-scheduled on 2021-07-26**
  #4514
* [nearcore 1.20.0-rc.1](https://github.com/near/nearcore/releases/tag/1.20.0-rc.1) was released and deployed to testnet
* [nearcore 1.20.0-rc.2](https://github.com/near/nearcore/releases/tag/1.20.0-rc.2) was released and deployed to testnet
* [nearcore 1.19.2](https://github.com/near/nearcore/releases/tag/1.19.2) was released and deployed to mainnet
* [NEAR Indexer for Explorer 0.7.0](https://github.com/near/near-indexer-for-explorer/releases/tag/0.7.0) compatible with nearcore 1.20.0-rc.1 is released
* [NEAR Indexer for Explorer 0.7.1](https://github.com/near/near-indexer-for-explorer/releases/tag/0.7.0) compatible with nearcore 1.20.0-rc.2 is released
* Explorer received Chinese localization 

# Results:

* explorer
  * **Added internationalization (i18n) and Chinese language** and resolved a few follow-up issues (kudos to @IsQiao @think-in-universe :muscle:)
    https://github.com/near/near-explorer/pull/684
    https://github.com/near/near-explorer/pull/679
  * Drafted the implementation to expose fees in the last 24 hours and 7 days through API (@shelegdmitriy)
    https://github.com/near/near-explorer/issues/627
  * Removed nearkat logo from footer (@shelegdmitriy)
    https://github.com/near/near-explorer/pull/678
* indexer-for-explorer
  * NEAR Indexer for Explorer 0.7.0 compatible with nearcore 1.20.0-rc.1 is released (@khorolets)
     [Release 0.7.0 · near/near-indexer-for-explorer · GitHub](https://github.com/near/near-indexer-for-explorer/releases/tag/0.7.0)
  * NEAR Indexer for Explorer 0.7.1 compatible with nearcore 1.20.0-rc.2 is released (@khorolets)
     [Release 0.7.1 · near/near-indexer-for-explorer · GitHub](https://github.com/near/near-indexer-for-explorer/releases/tag/0.7.1)
  * Null-bytes issue with method_names in AddKey action is fixed (@khorolets)
    https://github.com/near/near-indexer-for-explorer/pull/131
* indexer
  * Investigated how store fix can affect indexer https://github.com/near/nearcore/pull/4466. Found out that affected data is not from canonical chain and indexer does nothing about it (@khorolets)
  * R&D for auth system in Kafka for streaming indexer is being explored (@khorolets)
* nearcore
  * We had a new testnet release on 5th of July :tada: 
    https://github.com/near/nearcore/releases/tag/1.20.0-rc.1
  * We had a new mainnet release on 7th of July :tada:
    https://github.com/near/nearcore/releases/tag/1.19.2
  * Added the `--download-config` and `--boot-nodes` flags to the `neard init` command (@chefsale)
    https://github.com/near/nearcore/pull/4427
    https://github.com/near/nearcore/pull/4453
  * Drafted the refactoring to strictly-typed version of AccountId (@miraclx)
    https://github.com/near/nearcore/pull/4440
* nearup
  * Implemented support for guildnet in nearup (@chefsale)
    https://github.com/near/nearup/pull/181
* core-contracts
  * Finished the work on improving lockup contract with a help of the report from @ailisp https://hackmd.io/@ailisp/rJbiDYr5_ (@telezhnaya)
    https://github.com/near/core-contracts/pull/153
    https://github.com/near/core-contracts/pull/160
    https://github.com/near/core-contracts/pull/161
    https://github.com/near/core-contracts/pull/156
  * created a CI setup to ensure Wasm binaries are uploaded with according code changes and tests are run across contracts (@chefsale)
    https://github.com/near/core-contracts/pull/159
* near-sdk-rs:
  * Updated Dockerfile for building the contracts (@telezhnaya)
    https://github.com/near/near-sdk-rs/pull/454
* docs
  * Blockchain data flow is in beginning state of draft (thinking about schemas and graphics) (@khorolets)
  * Fixed general documentation describing lockups (@telezhnaya)
    https://github.com/near/docs/pull/775
  * Fixed doc about FT & NFT (@telezhnaya)
    https://github.com/near/NEPs/pull/236
* meta
  * Supported partners with Indexer (nearcore) release 1.20.0-rc.1 and later 1.20.0-rc.2 (@khorolets)

# Challenges & Blockers:

None

# Action Items:

* docs
  * Document blockchain data flow (@khorolets)
* operations
  * Wallet cloudflare integration to enable China usage (@mhalambek)
  * Intiate split of RPC services per product, example wallet has it's own JSON RPC endpoint and machines, same for bridge and so forth (@mhalambek)
  * Figure out costs of running RPC infrastructure (@chefsale)
  * Migrate CI deploy scripts to ansible (@chefsale)
* nearcore
  * Publish `near-account-id` as an independent crate to be used throughout the ecosystem. (@miraclx) (blocked while waiting for https://github.com/near/nearcore/pull/4440 review)
  * Work on getting reusable core crates (e.g `near-primitives`, `near-crypto`) lightweight and publishable (@miraclx)
  * Benchmark JSON-RPC performance in terms of requests per second (we focus on `query` method as others are considered as not very important to measure) (@khorolets)
  * Investigate broken backward compatible test https://github.com/near/nearcore/issues/4464 to find out if it was actually broken (@khorolets @frol)
* near-sdk-rs
  * Add the check to CI that all the binaries correspond to the source code (@telezhnaya, @chefsale)
    https://github.com/near/near-sdk-rs/pull/461
* indexer-for-explorer
  * Think how to provide right order of balance changes to the users in DB (@telezhnaya)
    https://github.com/near/near-indexer-for-explorer/issues/77#issuecomment-876441384
  * Calculate historical circulating supply and store it in indexer DB (@telezhnaya)
    https://github.com/near/near-indexer-for-explorer/commits/olya/circulating
  * Collect user requirements in regards to FT & NFT support on Indexer side (@telezhnaya)
    https://github.com/near/near-indexer-for-explorer/issues/108
  * Introduce logic to stop retrying to store data after N attempts and report the situation somewhere we can find it and handle (@khorolets)

**Comment by maxzaver** [comment]
Btw, did we have any feature requests for streaming indexer or was it something that we decided to implement based on our own judgement?

**Reply by khorolets** [reply]
We haven't any feature requests especially in this regard. This streaming indexer idea came around after we've realised the pain of syncing data, dealing with backups, speaking with partners about mentioned pains. We thought that such indexer might decrease the pain.

So yeah, it's more like "based on our own judgement"

**Reply by frol** [reply]
We had a number of requests starting from #1996, and then almost every project out there requesting events or "TheGraph experience". The solution we have in mind (Kafka-based) does not involve huge investments on the implementation side, but we don't want to fall into providing services for free forever, so we want to dogfood NEAR economy model and test it with our current Indexer Framework partners (they struggle to maintain Indexer nodes themselves due to lack of resources).

**Reply by frol** [reply]
Here we go, one more explicit request: https://github.com/near/nearcore/issues/4462#issuecomment-875131507
